rrule/src/core/rrule.rs

```diff
@@ -210,9 +175,7 @@ fn weekday_to_str(d: Weekday) -> String {
 /// It has two stages, based on the attached type, `Validated` or `Unvalidated`.
 /// - `Unvalidated`, which is the raw string representation of the RRULE
 /// - `Validated`, which is when the RRule has been parsed and validated, based on the start date
-#[cfg_attr(feature = "serde", serde_as)]
 #[derive(Debug, Clone, PartialEq, Eq)]
-#[cfg_attr(feature = "serde", derive(DeserializeFromStr, SerializeDisplay))]
 pub struct RRule<Stage = Validated> {
```

Comment:
It is the same reason as why why have to remove `FromStr` implementation for `RRule` as explained in the description. Is there a reason why someone would go from a text input to an `RRule` rather than an `RRuleSet`?

Reply:
Oh, I missed the description 🙈 

Cannot we change the `UNTIL` timezone, after validation? Or return error if timezone of `UNTIL` is different from timezone of `DTSTART`?

PS: I exactly use `RRule::from_str()` in our application :D

Reply:
Hmm, interesting. Can you provide some more details on how you use it and why you can't use `RRuleSet`? Is it a user input where you don't know the start date?

It seems like most other rrule implementations don't really follow the `UNTIL` / `DTSTART` sync according to the RFC, so I am kind of open to ignoring it if it proves to be a big pain to implement.

Reply:
I cannot provide more details, but yes, something like that, the input is like just the `RRule` line, without the `DTSTART`.

Reply:
<details>
<summary>Here is how the RFC defines `UNTIL` and how it should sync with `DTSTART`</summary>
      The UNTIL rule part defines a DATE or DATE-TIME value that bounds
      the recurrence rule in an inclusive manner.  If the value
      specified by UNTIL is synchronized with the specified recurrence,
      this DATE or DATE-TIME becomes the last instance of the
      recurrence.  The value of the UNTIL rule part MUST have the same
      value type as the "DTSTART" property.  Furthermore, if the
      "DTSTART" property is specified as a date with local time, then
      the UNTIL rule part MUST also be specified as a date with local
      time.  If the "DTSTART" property is specified as a date with UTC
      time or a date with local time and time zone reference, then the
      UNTIL rule part MUST be specified as a date with UTC time.  In the
      case of the "STANDARD" and "DAYLIGHT" sub-components the UNTIL
      rule part MUST always be specified as a date with UTC time.  If
      specified as a DATE-TIME value, then it MUST be specified in a UTC
      time format.  If not present, and the COUNT rule part is also not
      present, the "RRULE" is considered to repeat forever.
</details>

The key thing is that the `VALUE` (i.e. either `DATE` or `DATETIME`) parameter needs to be synced and that if `DTSTART` is a `DATE` with local time then `UNTIL` must also be in local time. The validator phase only sees the `DateTime` type for `until` and `dtstart` and cannot based on that type validate those requirements (`DateTime` does not tell us if the original text representation was in local time nor if it was a `DATE` originally). It should be possible to reorganize the code such that the we can pass this information to the validator, but I feel that it would be non-trivial and also add tech debt.

I understand that you can't share details about your company's code, but I can't really understand why the `impl FromStr for RRule` is useful. Could you either add a dummy start date to the text input or store the input as a string until you have access to the start date?

Reply:
OK, it's fine.
Most probably you are right, it's better to touch some bits in the app.